core/modules/entity/entity.module

```diff
@@ -20,6 +20,192 @@ function entity_modules_disabled() {
+  config_set('entity.view_modes', 'view_modes', $view_modes);
+
+  // Allow modules to respond after the view mode is saved.
+  if ($view_mode['is_new']) {
+    module_invoke_all('entity_view_mode_insert', $view_mode, $entity_type);
+  }
+  else {
+    // If the view mode's machine name has changed, invoke a separate hook
+    // before running the update hook.
+    if ($view_mode_name != $existing_view_mode_name) {
+      module_invoke_all('entity_view_mode_rename', $entity_type, $existing_view_mode_name, $view_mode_name);
+    }
+    module_invoke_all('entity_view_mode_update', $view_mode, $entity_type);
+  }
+
+  // Clear internal properties.
+  unset($view_mode['original']);
+  unset($view_mode['is_new']);
+
+  // Clear the static entity info cache and rebuild the menu.
+  entity_info_cache_clear();
+  state_set('menu_rebuild_needed', TRUE);
+}
+
+/**
+ * Delete a custom view mode.
+ *
+ * @param (string) $entity_type
+ *   Entity type. For example, node, user or term.
+ * @param  (string) $machine_name
+ *   Machine name of view mode.
+ */
+function entity_view_mode_delete($entity_type, $machine_name) {
+  if ($view_mode = entity_view_mode_load($entity_type, $machine_name)) {
+    module_invoke_all('entity_view_mode_delete', $view_mode, $entity_type);
+
+    $config = config('entity.view_modes');
+    $view_modes = $config->get('view_modes');
+    unset($view_modes[$entity_type][$machine_name]);
+    $config->set('view_modes', $view_modes);
+    $config->save();
+
+    // Clear the static entity info cache and rebuild the menu.
+    entity_info_cache_clear();
+    state_set('menu_rebuild_needed', TRUE);
+  }
+}
+
+/**
+ * Implements hook_entity_view().
+ *
+ * @see entity_preprocess()
+ */
+function entity_entity_view($entity, $entity_type, $view_mode, $langcode) {
+  // Add context and variables necessary to entity_preprocess().
+  list($id, $vid, $bundle) = entity_extract_ids($entity_type, $entity);
+  $entity->content['#entity_view_mode'] = array(
+    'entity_type' => $entity_type,
+    'id' => $id,
+    'bundle' => $bundle,
+    'view_mode' => $view_mode,
+    'langcode' => $langcode,
+    'has_bundles' => TRUE,
+  );
+  if ($entity_type == $bundle) {
+    $info = entity_get_info($entity_type);
+    if (empty($info['entity keys']['bundle'])) {
+      $entity->content['#entity_view_mode']['has_bundles'] = FALSE;
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess().
+ *
+ * Add template suggestions for an entity based on view modes if they do not
+ * already exist.
+ *
+ * The eventual order of the entity's theme suggestions are:
+ * - entity-type__bundle (lowest priority = first in the array)
+ * - entity-type__bundle__view-mode
+ * - entity-type__id
+ * - entity-type__id__view-mode (highest priority = last in the array)
+ *
+ * @see entity_view_mode_entity_view()
+ */
+function entity_preprocess(&$variables, $hook) {
+  // Check for the context provided in entity_view_mode_entity_view().
```

Comment:
added suggestions for comment, node, and term and user. Hope I didn't miss any entities.